### PR TITLE
Added last changed by user

### PR DESCRIPTION
**Pull request by jnimmo**
Thanks for the pointers that made it very easy.
Tested this morning and working correctly. Are you able to look at doing a release for it or anything you want me to tidy up? Cheers

### CODE REVIEW

File: pyenvisalink/dsc_client.py [hunk 1]
```diff
@@ -159,7 +159,12 @@ def handle_partition_state_change(self, code, data):
                 partitionNumber = int(data[0])
                 self._alarmPanel.alarm_state['partition'][partitionNumber]['status'].update(evl_ResponseTypes[code]['status'])
                 _LOGGER.debug(str.format("(partition {0}) state has updated: {1}", partitionNumber, json.dumps(evl_ResponseTypes[code]['status'])))
-                return partitionNumber
+                '''Log the user who last armed or disarmed the alarm'''
+                if code in ['700','750']:
+                    lastChangedByUser = {'last_changed_by_user': int(data[1:5])}
+                    self._alarmPanel.alarm_state['partition'][partitionNumber]['status'].update(lastChangedByUser)
+
+		return partitionNumber
```
Comment (jnimmo) [inline pyenvisalink/dsc_client.py]: Sorry dodgy spacing here fixed in the next commit

File: pyenvisalink/alarm_state.py [hunk 2]
```diff
@@ -13,7 +13,7 @@ def get_initial_alarm_state(maxZones, maxPartitions):
                                                       'armed_zero_entry_delay': False, 'alarm_fire_zone': False,
                                                       'trouble': False, 'bat_trouble': False, 'ready': False, 'fire': False,
                                                       'armed_stay': False, 'alpha': 'N/A', 'beep': False,
-                                                      'exit_delay': False, 'entry_delay': False,}}
+                                                      'exit_delay': False, 'entry_delay': False, 'last_changed_by_user': ''}}
```
Comment (Cinntax) [inline pyenvisalink/alarm_state.py]: What do you think about a last disarmed by user + last armed by user variables?

File: pyenvisalink/dsc_client.py [hunk 3]
```diff
@@ -30,9 +30,8 @@ def dump_zone_timers(self):
         self.send_command(evl_Commands['DumpZoneTimers'], '')
 
     def keypresses_to_partition(self, partitionNumber, keypresses):
-        """Send keypresses to a particular partition."""
-        for char in keypresses:
-            self.send_command(evl_Commands['PartitionKeypress'], str.format("{0}{1}", partitionNumber, char))
+        """Send keypresses (max of 6) to a particular partition."""
+	self.send_command(evl_Commands['PartitionKeypress'], str.format("{0}{1}", partitionNumber, keypresses[:6]))
```
Comment (Cinntax) [inline pyenvisalink/dsc_client.py]: I know this is incredibly nit-picky, but the indentation of the "self.send_command". doesn't match the comment above. Can you space it in 1 more?

**Review by Cinntax — CHANGES_REQUESTED** [review]

**Comment by Cinntax** [comment]
Thanks for your help on this!  A couple small things- I'll then do a release.

**Comment by jnimmo** [comment]
Good spotting, hadn't figured out Git on desktop and the web editor was doing some funky stuff with tabs..
I've pulled it all back into desktop and de-tabbed so should be good now.
Also split into last_armed_by_user etc as that makes sense
Now I'm just not sure if you've got my latest code or if I have to somehow merge it - desktop is talking about a conflict.

**Comment by Cinntax** [comment]
Shoot- doesn't look like I do have the latest code. I think you will have to do some merging on your end before I can on mine...

**Comment by jnimmo** [comment]
Ok looks like I managed to merge it now sorry
Instead of if code in ['700','750'] you should now see
if code == '700':
....
elif code == '750':
and those other changes tidied up

**Review by Cinntax — APPROVED** [review]
Excellent- thanks!!  Are you ready for a release then?  I'm guessing so.

**Comment by jnimmo** [comment]
Thanks very much, yeah ready for a release then I'll properly test out the HA changes - not sure what the requirements are for testing/lint etc but I guess I'll find out!

**Comment by Cinntax** [comment]
Okay- 1.9 has been released!  Go ahead and reference that in the hass component.

**Comment by jnimmo** [comment]
Awesome thanks, added that and mine has updated and working well

**Pull request merged**